lib/sip2/responses/base_response.rb

```diff
@@ -10,7 +10,10 @@ class BaseResponse
       @@response_objects = {}
 
       def initialize(raw_response)
-        @raw_response = raw_response.strip
+        @raw_response = raw_response&.strip
+        if @raw_response.nil? || @raw_response.empty?
+          raise ArgumentError, 'raw_response from SIP2 server cannot be blank'
```

Comment:
We sometimes get this situation when the SIP2 server has its whitelist misconfigured, or some other configuration on the remote server is wrong.  It would be useful if we can propagate that specific situation up the chain.  Perhaps a custom error (EmptyResponseError or something similar?).

Alternatively we could catch this situation at `Connection#send_and_handle_message`.  What do you think?

Reply:
I've replaced it with a custom error class. That's probably a better idea regardless of where/how we want to catch it later.